### DISCUSSION TITLE
Ability to have flip in recording settings

### DISCUSSION THREAD

**Discussion by dogukandemir** [obsproject/obs-studio#7097] [General]
I would like to start a discussion about having an ability to flip recordings. Motivation of this discussion started with using virtual camera to present on discord and record that presentation at the same time.

When I use virtual camera and flip it horizontally, audiences can see presentation in a normal way. If I record it at the same time, recording becomes horizontally flipped and I need to use another software to flip that recording back to normal.

I think it would be really nice to have a functionality to flip recordings for people who wants to use OBS for the scenario that I mentioned above.

**Comment by Nofew** [comment]
Actually, the preview in discord is only flipped for you. Everyone else sees it normally. If you flip it so the preview appears correct to you, then everyone else sees it backwards, just like when you record right now and see it backwards.

**Comment by dogukandemir** [comment]
Looks like it's totally my misunderstanding of discord preview. Sorry for taking your time :)

**Comment by Fenrirthviti** [comment]
In the future, we recommend asking about stuff like this on our Discord, or on the Ideas page. As mentioned by the [stickied post](https://github.com/obsproject/obs-studio/discussions/5206) we don't want to use github discussions for feature requests.

**Comment by hubyhuby** [comment]
I have exactely the same problem all my videos recordings are flipped.
How to reproduce :
1. WHen creating a texte for example in OBS. Your writing is from left to right normal.
2.  When you get in google meet it is flipped
3. So you mirror everything in OBS. Meaning you cannot read it yourself anymore in OBS, but in google meet its working for you and the people you are talking to.
4. Then you start recording at the same time as you stream on google meet. But when you look at the recordings they are like what you look at in OBS.... flipped horizontaly.  So you cannot read the textes.

Basically you want to record what you send toyour google meet viewers.

Made a script, if it can help people around this bug : 
https://github.com/hubyhuby/linux-scripts/blob/main/mirror_obs_webm.sh

UPDATE :
Apparentely I didn t need to flip horizontaly, you are rihgt . I think now it will all be fine.
I will confirme so we can close this issue

Just did a video and it works.
You are right , no need to flip horizontaly. I am wondering why I was doing this before... Anyhow with OBS today its works this way. No need to flip all the ressources in OBS.

**Reply by Fenrirthviti** [reply]
As mentioned above, nothing is flipped on the OBS side, and you shouldn't be flipping anything there. What you are seeing is the Google Meet preview flipping the virtual camera, and the flip is only happening for you. If you manually flip it to try and make the preview look "correct" then everyone else will see it flipped. What shows up in the OBS preview is what will be output, there is no flipping otherwise.
